apps/vercel/frontend/src/clients/Vercel.ts

```diff
@@ -87,43 +87,51 @@ export default class VercelClient implements VercelAPIClient {
     return formattedApiPaths;
   }
 
-  async getLatestDeploymentId(projectId: string): Promise<string> {
-    const res = await fetch(`${this.baseEndpoint}/v6/deployments?projectId=${projectId}`, {
-      headers: this.buildHeaders(),
-      method: 'GET',
-    });
-
-    const data = await res.json();
-
-    // Vercel returns deployments sorted by date in descending order
-    return data.deployments[0].uid;
-  }
-
-  async createDeployment({ project }: CreateDeploymentInput): Promise<Deployment> {
-    const res = await fetch(`${this.baseEndpoint}/v13/deployments`, {
-      headers: this.buildHeaders(),
-      method: 'POST',
-      body: JSON.stringify({
-        name: project.name,
-        deploymentId: project.targets.production.id,
-        target: 'production',
-      }),
-    });
+  async listDeploymentSummary(
+    projectId: string,
+    teamId?: string,
+    deploymentId?: string
+  ): Promise<ListDeploymentSummaryResponse> {
+    let data: ListDeploymentSummaryResponse = { serverlessFunctions: [] };
+    try {
+      const latestDeploymentId =
+        deploymentId || (await this.getLatestDeploymentId(projectId, teamId));
+      const res = await fetch(
+        `${
+          this.baseEndpoint
+        }/v6/deployments/${latestDeploymentId}/files/outputs?file=..%2Fdeploy_metadata.json&${this.buildTeamIdQueryParam(
+          teamId
+        )}`,
+        {
+          headers: this.buildHeaders(),
+          method: 'GET',
+        }
+      );
 
-    const data = await res.json();
+      data = await res.json();
+    } catch (e) {
+      console.error(e);
+      throw new Error('Failed to fetch deployment summary.');
+    }
 
     return data;
   }
 
-  async getDeploymentById(deploymentId: string): Promise<Deployment> {
-    const res = await fetch(`${this.baseEndpoint}/v13/deployments/${deploymentId}`, {
-      headers: this.buildHeaders(),
-      method: 'GET',
-    });
+  async getLatestDeploymentId(projectId: string, teamId?: string): Promise<string> {
```

Comment:
maybe we want this to be called `getLatestDeploymentById` (include the word "By')

Reply:
oh good idea - this here is grabbing the latest deployment ID itself, so not the latest deployment by ID. Though maybe I will make it more generically return the latest deployment, and the consuming method can grab the ID as it sees fit 👍 will adjust!